galley/pkg/runtime/publish/strategy.go

```diff
@@ -126,26 +120,18 @@ func (s *Strategy) OnChange() {
 
 // startTimer performs a start or reset on the timer. Called with lock on stateLock.
 func (s *Strategy) startTimer() {
-	if s.stopped {
-		panic("asyncTimer attempting to restart stopped timer")
-	}
-
 	if s.timer != nil {
-		// Already started, just perform a reset of the timer.
-		s.resetChan <- struct{}{}
+		// Already started, just perform a non-blocking reset of the timer.
+		select {
+		case s.resetChan <- struct{}{}:
+		default:
```

Comment:
It looks like this is the main functional change, right?

Reply:
It was another similar race to Close, but I don't think this was the culprit. I've just updated this logic so that we're not holding onto the state lock when we send the reset (thus avoiding the need for a non-blocking write).

The real issue, I believe, was that `Close` was holding the state lock when sending on the stop channel previously.  Now that we've switched over to `Worker` that's no longer the case (it uses it's own locks).